packages/loader/container-loader/src/loader.ts

```diff
@@ -455,6 +460,35 @@ export class Loader implements IHostLoader {
         return { container, parsed };
     }
 
+    private async resolveWithDomainChangeHandling(
+        request: IRequest,
+        pendingLocalState?: IPendingLocalState,
+    ): Promise<{ container: Container; parsed: IParsedUrl }> {
+        let req: IRequest = request;
+        let success = false;
+        let response;
+        do {
+            const resolvedUrl = await this.resolveUrl(req);
+            try {
+                response = await this.getContainer(resolvedUrl, req, pendingLocalState);
+                success = true;
```

Comment:
Instead of having success variable, I'd break out of here, and use while(true) or for(;;) loop instead.